openupgrade_scripts/scripts/website_sale/14.0.1.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,89 @@
+---Models in module 'website_sale'---
+obsolete model product.style (renamed to product.ribbon)
+new model product.ribbon (renamed from product.style)
+# DONE: pre-migration: renamed models and tables
+
+new model website.sale.extra.field
+# NOTHING TO DO
+
+---Fields in module 'website_sale'---
+website_sale / product.public.category  / _order                        : _order is now 'sequence, name, id' ('sequence, name')
+website_sale / product.public.category  / seo_name (char)               : NEW
+# NOTHING TO DO
+
+website_sale / product.style            / name (char)                   : DEL required
+website_sale / product.ribbon           / html (char)                   : NEW required
+# DONE: pre-migration: renamed fields
+
+website_sale / product.style            / html_class (char)             : now required, req_default: function
+# DONE: pre-migration: set default '' in case is NULL
+
+website_sale / product.ribbon           / bg_color (char)               : NEW
+website_sale / product.ribbon           / text_color (char)             : NEW
```

Comment:
Shouldn't this preserved the old colors? What happens if not filled?

Reply:
These are new fields, new features, for more customization. The old colors were defined by class, so they should still be in the html_class.
![Selection_501](https://user-images.githubusercontent.com/25005517/181747233-1e13e573-7b32-4246-9b2c-5d4adbc77a3c.png)
https://github.com/odoo/odoo/commit/c058f19bfa33b510addfed314aa27d7e5c7fdb18